### PR TITLE
Merge master post 4.7

### PR DESCRIPTION
**Pull request by Ericgig**
**Description**

Merge the latest changes of master.
- Activating the windows tests.
- Updating the changelog.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -77,6 +77,16 @@ jobs:
...
 
+          # Windows. Once all tests pass without special options needed, this
+          # can be moved to the main os list in the test matrix. All the tests
+          # that fail currently seem to do so because mcsolve uses
+          # multiprocessing under the hood. Windows does not support fork()
+          # well, which makes transfering objects to the child processes
+          # error prone. See, e.g., https://github.com/qutip/qutip/issues/1202
+          - case-name: Windows Latest
+            os: windows-latest
+            pytest-extra-options: "-k 'not (test_correlation or test_interpolate or test_mcsolve)'"
```
Comment (hodgestar) [inline .github/workflows/tests.yml:88]: We might need to also skip `test_parallel` on v5 and leave it for fixing later.

**Review by hodgestar — APPROVED** [review]
The merge look good. We should make tests pass before landing the PR though.

**Review by hodgestar — COMMENTED** [review]
Left some questions (mostly about Windows).

File: qutip/solver/parallel.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
     Pool = multiprocessing.Pool
 
 map_kw = {
-    'job_timeout': 1e8,
+    'job_timeout': 2e16-1,
```
[the remark this answers is not in the document]
Comment (Ericgig) [reply qutip/solver/parallel.py]: I meant 2**16-1, but there is `threading.TIMEOUT_MAX` which should always work as intended.

File: qutip/core/coefficient.py [hunk 3]
```diff
@@ -346,7 +346,10 @@ def coeff_from_str(base, args, args_ctypes, compile_opt=None):
     if coeff is None or compile_opt['recompile']:
         code = make_cy_code(parsed, variables, constants,
                             raw, compile_opt)
-        coeff = compile_code(code, file_name, parsed, compile_opt)
+        try :
+            coeff = compile_code(code, file_name, parsed, compile_opt)
+        except Exception:
+            return StrFunctionCoefficient(base, args)
```
[the remark this answers is not in the document]
Comment (hodgestar) [reply qutip/core/coefficient.py:352]: Can we add a TODO to see if we can find the source of the PermissionError later?

**Pull request merged**